base/reduce.jl

```diff
@@ -491,60 +498,82 @@ In the former case, the integers are widened to system word size and therefore
 the result is 128. In the latter case, no such widening happens and integer
 overflow results in -128.
 """
-sum(f, a) = mapreduce(f, add_sum, a)
+sum(f, a; kw...) = mapreduce(f, add_sum, a; kw...)
```

Comment:
My thought was that some non-array collection might want to implement `dims`.

Reply:
Another issue is that passing `init` breaks something like `sum(inv, 1)` because `mapreduce(f, op, a::Number)` does not take any keyword arguments

https://github.com/JuliaLang/julia/blob/7301dc61bdeb5d66e94e15bdfcd4c54f7c90f068/base/reduce.jl#L419-L420

It also breaks collection that dispatches to external `mapreduce` implementation that does not use `Base._InitialValue`.

Reply:
OK, that's fine then.